src/main/java/io/vertx/rabbitmq/RabbitMQClient.java

```diff
@@ -145,6 +174,51 @@ default void basicConsumer(String queue, Handler<AsyncResult<RabbitMQConsumer>>
   @GenIgnore(GenIgnore.PERMITTED_TYPE)
   Future<Void> basicPublish(String exchange, String routingKey, BasicProperties properties, Buffer body);
 
+  /**
+   * Publish a message. Publishing to a non-existent exchange will result in a channel-level protocol exception,
+   * which closes the channel. Invocations of Channel#basicPublish will eventually block if a resource-driven alarm is in effect.
+   *
+   * @param deliveryTagHandler callback to capture the deliveryTag for this message.  Note that this will be called synchronously in the context of the client.
+   * @see com.rabbitmq.client.Channel#basicPublish(String, String, AMQP.BasicProperties, byte[])
+   */
+  @GenIgnore(GenIgnore.PERMITTED_TYPE)
+  void basicPublishWithDeliveryTag(String exchange, String routingKey, BasicProperties properties, Buffer body, @Nullable Handler<Long> deliveryTagHandler, Handler<AsyncResult<Void>> resultHandler);
```

Comment:
remove `deliveryTagHandler` and instead have resultHandler be a `Handler<List<Long>>`

Reply:
The delivery tag needs to be known before the result is complete.
The confirmation may be sent before the resultHandler gets called.

I can expand the javadoc if that would make this clearer.

Reply:
Added the comment:
```
   * The deliveryTagHandler will be called before the message is sent, which is necessary because the confirmation may arrive asynchronously before the resultHandler is called.
```

Reply:
it can be known before, however that does not prevent the resultHandler to be called I think, i.e is there an interaction that can be triggered by the deliveryTagHandler before the resultHandler is called that will have any side effect on the result ?